src/domain/usecases/metadata/ApplySharingSettingsUseCase.ts

```diff
@@ -37,15 +37,17 @@ export class ApplySharingSettingsUseCase implements UseCase {
     }
 
     private updateSharingSettings(item: MetadataItem, sharings: SharedObject, replace: boolean): MetadataItem {
```

Comment:
You need also to update `userGroups`, `users` and `public` inside new `sharing` object in every item (not only `publicAccess`, `userAccesses` and `userGroupAccesses` that are ok). In addition, you need to apply the function `assertValidSharingSettings` to `acces`  property inside properties in the new `sharing` object